src/elements/form/Control.ts

```diff
@@ -251,37 +259,51 @@ export class NovoControlElement extends OutsideClick implements OnInit, OnDestro
         this.formattedValue = this.labels.formatDateWithFormat(event.date, this.labels.dateTimeFormat);
     }
 
+
     resizeTextArea(event) {
         // Reset the height
         let height = event.target.value.length > 0 ? `${event.target.scrollHeight}px` : '2rem';
         event.target.style.height = '';
         event.target.style.height = height;
     }
 
+    checkMaxLength(event) {
+        if (this.control && this.control.maxlength) {
+            this.characterCount = event.target.value.length;
+            if (event.target.value.length >= this.control.maxlength) {
+                this.maxLengthMet = true;
+            } else {
+                this.maxLengthMet = false;
+            }
+        }
+    }
+
     modelChange(value) {
         if (!value) {
             this._focused = false;
         }
         this.change.emit(value);
+        this.checkMaxLength(value);
```

Comment:
`modelChange` is only used on Picker/Tiles/Chips, don't think you need to check it here. You only care about textbox and textarea